nixos/modules/services/continuous-integration/github-runner/options.nix

```
@@ -170,4 +170,14 @@ with lib;
     default = null;
     defaultText = literalExpression "username";
   };
+
+  workDir = mkOption {
+    type = types.nullOr types.str;
+    description = lib.mdDoc ''
+      Working directory, used for repository checkouts.
```

Just a suggestion for some additional information:

```suggestion
      Working directory, available as `$GITHUB_WORKSPACE` during workflow runs and used as a default for [repository checkouts](https://github.com/actions/checkout). The service cleans this directory on every service start.
```

Maybe it's also worth mentioning that the `github-runner` user needs permission to access this directory.